### ISSUE TITLE
salt execution module junos.install_config with mode dynamic throws exception RpcError: RpcError(severity: error, bad_element: None, message: error: statement creation failed: version

### ISSUE THREAD

**Issue by rsmekala** [Juniper/salt#104]
### Description of Issue/Question

```
set policy-options policy-statement test term test from route-filter 10.0.1.0/24 address-mask 255.0.255.0
set policy-options policy-statement test then accept

root@nms5-salt-master-b:/srv/salt# cat config_dynamic.xml 
<policy-options>
                <policy-statement>
                    <name>test</name>
                    <term>
                        <name>test</name>
                        <from>
                            <route-filter>
                                <address>10.0.1.0/24</address>
                                <address-mask>255.0.255.0</address-mask>
                            </route-filter>
                        </from>
                    </term>
                    <then>
                        <accept/>
                    </then>
                </policy-statement>
            </policy-options>




root@nms5-salt-master-b:/srv/salt# salt "vmx1" junos.install_config salt://config_dynamic.set mode=dynamic comment="commit dynamic mode" format=set
[WARNING ] /usr/local/lib/python2.7/site-packages/salt/payload.py:149: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  ret = msgpack.loads(msg, use_list=True, ext_hook=ext_type_decoder, encoding=encoding)

vmx1:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/local/lib/python2.7/site-packages/salt/minion.py", line 1660, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "/usr/local/lib/python2.7/site-packages/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/salt/modules/junos.py", line 99, in wrapper
        return function(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/salt/modules/junos.py", line 935, in install_config
        config_diff = cu.diff()
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/utils/config.py", line 228, in diff
        compare='rollback', rollback=str(rb_id), format='text'
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/rpcmeta.py", line 345, in _exec_rpc
        return self._junos.execute(rpc, **dec_args)
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/decorators.py", line 76, in wrapper
        return function(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/decorators.py", line 31, in wrapper
        return function(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/device.py", line 794, in execute
        errs=ex)
    RpcError: RpcError(severity: error, bad_element: None, message: error: statement creation failed: version
    error: /config/juniper.conf:5:(0) statement creation failed
    error: /config/juniper.conf:100:(9) error recovery ignores input until this point
    error: /config/juniper.conf:104:(9) error recovery ignores input until this point
    error: /config/juniper.conf:124:(9) error recovery ignores input until this point
    error: /config/juniper.conf:135:(9) error recovery ignores input until this point
    error: /config/juniper.conf:210:(9) error recovery ignores input until this point
    warning: mgd: statement must contain additional statements
    error: /config/juniper.conf:212:(0) statement creation failed
    error: /config/juniper.conf:216:(9) error recovery ignores input until this point
    error: /config/juniper.conf:225:(9) error recovery ignores input until this point
    warning: mgd: statement must contain additional statements
    error: /config/juniper.conf:227:(0) statement creation failed
    error: /config/juniper.conf:231:(9) error recovery ignores input until this point
    error: /config/juniper.conf:240:(9) error recovery ignores input until this point
    warning: mgd: statement must contain additional statements
    error: /config/juniper.conf:243:(0) statement creation failed
    error: /config/juniper.conf:265:(1) error recovery ignores input until this point
    error: /config/juniper.conf:275:(1) error recovery ignores input until this point)
ERROR: Minions returned with non-zero exit code
root@nms5-salt-master-b:/srv/salt# salt "vmx1" junos.install_config salt://config_dynamic.set mode=dynamic comment="commit dynamic mode"
```

**Comment by rsmekala** [comment]
**_Problem_**: Dynamic mode doesn't have diff, commit check operation support. 
**_Solution_**: If-else conditions

**Comment by rsmekala** [comment]
Merged to 2019.2
